.github/workflows/lint.yaml

```diff
@@ -0,0 +1,101 @@
+name: lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+      - release-*
+  pull_request:
+    branches:
+      - main
+      - release-*
+
+# cancel the in-progress workflow when PR is refreshed.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  codespell:
+    name: codespell
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: codespell
+        uses: codespell-project/actions-codespell@master
```

Comment:
let's stick to a version instead of a master when we have a new release branch and if we use master we need to fix all the checks (newly) added etc which is going to be not possible always.

Reply:
the specific tags/branch are not updated recently since years but master is updated 3 weeks ago, so I guess we have to use master also in rook we are using master also and seems quite stable